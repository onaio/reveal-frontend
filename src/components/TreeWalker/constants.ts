export const ACTIVE = 'Active';
export const FEATURE = 'Feature';
export const FIND_BY_ID = 'location/findByJurisdictionIds';
export const FIND_BY_PROPERTIES = 'location/findByProperties';
export const LOCATION = 'location';
export const COULDNT_LOAD_CHILDREN = 'Could not load children';
export const COULDNT_LOAD_PARENTS = 'Could not load parents';
export const NOT_AVAILABLE = 'N/A';
