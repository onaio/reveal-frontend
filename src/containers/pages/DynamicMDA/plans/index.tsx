import reducerRegistry from '@onaio/redux-reducer-registry';
import React from 'react';
import { connect } from 'react-redux';
import { RouteComponentProps } from 'react-router';
import { Store } from 'redux';
import {
  HIDDEN_PLAN_STATUSES,
  SUPERSET_DYNAMIC_MDA_REPORTING_PLANS_SLICE,
} from '../../../../configs/env';
import { MDA_PLANS } from '../../../../configs/lang';
import { QUERY_PARAM_TITLE, REPORT_MDA_PLAN_URL } from '../../../../constants';
import { getPlanStatusToDisplay, getQueryParams, RouteParams } from '../../../../helpers/utils';
import supersetFetch from '../../../../services/superset';
import GenericPLansReducer, {
  genericFetchPlans,
  makeGenericPlansArraySelector,
  reducerName as GenericPLansReducerName,
} from '../../../../store/ducks/generic/plans';
import { GenericPlan } from '../../../../store/ducks/generic/plans';
import { InterventionType } from '../../../../store/ducks/plans';
import { GenericPlanListProps, GenericPlansList } from '../../GenericPlansList';

/** register the MDA plan definitions reducer */
reducerRegistry.register(GenericPLansReducerName, GenericPLansReducer);

/** a list of plan statuses to be displayed */
const allowedPlanStatusList = getPlanStatusToDisplay(HIDDEN_PLAN_STATUSES);

/** selector for Dynamic MDA plans */
const MDAPlansArraySelector = makeGenericPlansArraySelector();

/** Simple component that loads a list of MDA plans */
const MDAPLansList = (props: GenericPlanListProps & RouteComponentProps) => {
  return (
    <div>
      <GenericPlansList {...props} />
    </div>
  );
};

/** Declare default props for MDAPLansList */
const defaultProps: GenericPlanListProps = {
  fetchPlans: genericFetchPlans,
  pageTitle: MDA_PLANS,
  pageUrl: REPORT_MDA_PLAN_URL,
  plans: [],
  service: supersetFetch,
  supersetReportingSlice: SUPERSET_DYNAMIC_MDA_REPORTING_PLANS_SLICE,
};

MDAPLansList.defaultProps = defaultProps;

export { MDAPLansList };

/** Connect the component to the store */

/** interface to describe props from mapStateToProps */
interface DispatchedStateProps extends RouteComponentProps<RouteParams> {
  plans: GenericPlan[];
}

/** map state to props */
const mapStateToProps = (
  state: Partial<Store>,
  ownProps: RouteComponentProps<RouteParams>
): DispatchedStateProps => {
  const searchedTitle = getQueryParams(ownProps.location)[QUERY_PARAM_TITLE] as string;
  const plans = MDAPlansArraySelector(state, {
    interventionTypes: [InterventionType.DynamicMDA],
    plan_title: searchedTitle,
    statusList: allowedPlanStatusList,
  });

  return {
    ...ownProps,
    plans,
  };
};

/** map dispatch to props */
const mapDispatchToProps = { fetchPlans: genericFetchPlans };

/** Connected MDA Plans component */
const ConnectedMDAPLansList = connect(mapStateToProps, mapDispatchToProps)(MDAPLansList);

export default ConnectedMDAPLansList;
