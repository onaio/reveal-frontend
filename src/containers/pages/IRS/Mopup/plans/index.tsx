import React from 'react';
import { RouteComponentProps } from 'react-router-dom';
import { MOP_UP_REPORTING_TITLE } from '../../../../../configs/lang';
import { IRS_MOP_UP_REPORT_URL } from '../../../../../constants';
import { RouteParams } from '../../../../../helpers/utils';
import ConnectedIRSPlansList from '../../plans';

export const IRSMopUpReporting = (props: RouteComponentProps<RouteParams>) => {
  const plansProps = {
    ...props,
    pageTitle: MOP_UP_REPORTING_TITLE,
    pageUrl: IRS_MOP_UP_REPORT_URL,
  };
  return <ConnectedIRSPlansList {...plansProps} />;
};
