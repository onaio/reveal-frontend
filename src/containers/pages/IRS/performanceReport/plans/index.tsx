import React from 'react';
import { RouteComponentProps } from 'react-router-dom';
import { IRS_PERFORMANCE_REPORTING_TITLE } from '../../../../../configs/lang';
import { PERFORMANCE_REPORT_IRS_PLAN_URL } from '../../../../../constants';
import { RouteParams } from '../../../../../helpers/utils';
import ConnectedIRSPlansList from '../../plans';

export const IRSPerfomenceReport = (props: RouteComponentProps<RouteParams>) => {
  const plansProps = {
    ...props,
    pageTitle: IRS_PERFORMANCE_REPORTING_TITLE,
    pageUrl: PERFORMANCE_REPORT_IRS_PLAN_URL,
  };
  return <ConnectedIRSPlansList {...plansProps} />;
};
