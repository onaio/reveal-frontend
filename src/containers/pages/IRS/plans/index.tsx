import ListView from '@onaio/list-view';
import reducerRegistry from '@onaio/redux-reducer-registry';
import React, { useEffect, useState } from 'react';
import { Helmet } from 'react-helmet';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { Col, Row } from 'reactstrap';
import { Store } from 'redux';
import HeaderBreadcrumb from '../../../../components/page/HeaderBreadcrumb/HeaderBreadcrumb';
import Loading from '../../../../components/page/Loading';
import { SUPERSET_IRS_REPORTING_PLANS_SLICE } from '../../../../configs/env';
import { HOME, HOME_URL, IRS_PLANS, REPORT_IRS_PLAN_URL } from '../../../../constants';
import supersetFetch from '../../../../services/superset';
import IRSPlansReducer, {
  fetchIRSPlans,
  getIRSPlansArray,
  IRSPlan,
  reducerName as IRSPlansReducerName,
} from '../../../../store/ducks/generic/plans';

/** register the plan definitions reducer */
reducerRegistry.register(IRSPlansReducerName, IRSPlansReducer);

/** interface for PlanList props */
interface PlanListProps {
  fetchPlans: typeof fetchIRSPlans;
  plans: IRSPlan[];
  service: typeof supersetFetch;
}

/** Simple component that loads the new plan form and allows you to create a new plan */
const IRSPlansList = (props: PlanListProps) => {
  const { fetchPlans, plans, service } = props;
  const [loading, setLoading] = useState<boolean>(true);

  const pageTitle: string = IRS_PLANS;

  const breadcrumbProps = {
    currentPage: {
      label: pageTitle,
      url: REPORT_IRS_PLAN_URL,
    },
    pages: [
      {
        label: HOME,
        url: HOME_URL,
      },
    ],
  };

  /** async function to load the data */
  async function loadData() {
    try {
      setLoading(plans.length < 1); // only set loading when there are no plans
      await service(SUPERSET_IRS_REPORTING_PLANS_SLICE).then((result: IRSPlan[]) =>
        fetchPlans(result)
      );
    } catch (e) {
      // do something with the error?
    } finally {
      setLoading(false);
    }
  }

  useEffect(() => {
    loadData();
  }, []);

  if (loading === true) {
    return <Loading />;
  }

  const listViewProps = {
    data: plans.map(planObj => {
      return [
        <Link to={`${REPORT_IRS_PLAN_URL}/${planObj.plan_id}`} key={planObj.plan_id}>
          {planObj.plan_title}
        </Link>,
        planObj.plan_date,
        planObj.plan_effective_period_start,
        planObj.plan_effective_period_end,
        planObj.plan_status,
      ];
    }),
    headerItems: ['Title', 'Date Created', 'Start Date', 'End Date', 'Status'],
    tableClass: 'table table-bordered plans-list',
  };

  return (
    <div>
      <Helmet>
        <title>{pageTitle}</title>
      </Helmet>
      <HeaderBreadcrumb {...breadcrumbProps} />
      <Row>
        <Col md={8}>
          <h3 className="mt-3 mb-3 page-title">{pageTitle}</h3>
        </Col>
      </Row>
      <Row>
        <Col>
          <ListView {...listViewProps} />
        </Col>
      </Row>
    </div>
  );
};

/** Declare default props for IRSPlansList */
const defaultProps: PlanListProps = {
  fetchPlans: fetchIRSPlans,
  plans: [],
  service: supersetFetch,
};

IRSPlansList.defaultProps = defaultProps;

export { IRSPlansList };

/** Connect the component to the store */

/** interface to describe props from mapStateToProps */
interface DispatchedStateProps {
  plans: IRSPlan[];
}

/** map state to props */
const mapStateToProps = (state: Partial<Store>): DispatchedStateProps => {
  const planDefinitionsArray = getIRSPlansArray(state);

  return {
    plans: planDefinitionsArray,
  };
};

/** map dispatch to props */
const mapDispatchToProps = { fetchPlans: fetchIRSPlans };

/** Connected ActiveFI component */
const ConnectedIRSPlansList = connect(
  mapStateToProps,
  mapDispatchToProps
)(IRSPlansList);

export default ConnectedIRSPlansList;
