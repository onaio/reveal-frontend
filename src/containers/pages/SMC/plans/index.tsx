import reducerRegistry from '@onaio/redux-reducer-registry';
import React from 'react';
import { connect } from 'react-redux';
import { RouteComponentProps } from 'react-router';
import { Store } from 'redux';
import { HIDDEN_PLAN_STATUSES, SUPERSET_SMC_REPORTING_PLANS_SLICE } from '../../../../configs/env';
import { SMC_PLANS } from '../../../../configs/lang';
import { QUERY_PARAM_TITLE, REPORT_SMC_PLAN_URL } from '../../../../constants';
import { getPlanStatusToDisplay, getQueryParams } from '../../../../helpers/utils';
import { RouteParams } from '../../../../helpers/utils';
import supersetFetch from '../../../../services/superset';
import SMCPlansReducer, {
  fetchSMCPlans,
  makeSMCPlansArraySelector,
  reducerName as SMCPlansReducerName,
  SMCPLANType,
} from '../../../../store/ducks/generic/SMCPlans';
import { GenericPlanListProps, GenericPlansList } from '../../GenericPlansList';

/** register the SMC plan definitions reducer */
reducerRegistry.register(SMCPlansReducerName, SMCPlansReducer);

/** a list of plan statuses to be displayed */
const allowedPlanStatusList = getPlanStatusToDisplay(HIDDEN_PLAN_STATUSES);
// store.dispatch(fetchSMCPlans(testlans as any))

/** Simple component that loads a preview list of SMC plans */
const SMCPlansList = (props: GenericPlanListProps & RouteComponentProps) => {
  return (
    <div>
      <GenericPlansList {...props} />
    </div>
  );
};

/** Declare default props for SMCPlansList */
const defaultProps: GenericPlanListProps = {
  fetchPlans: fetchSMCPlans,
  pageTitle: SMC_PLANS,
  pageUrl: REPORT_SMC_PLAN_URL,
  plans: [],
  service: supersetFetch,
  supersetReportingSlice: SUPERSET_SMC_REPORTING_PLANS_SLICE,
};

SMCPlansList.defaultProps = defaultProps;

export { SMCPlansList };

/** Connect the component to the store */

/** interface to describe props from mapStateToProps */
interface DispatchedStateProps extends RouteComponentProps<RouteParams> {
  plans: SMCPLANType[];
}

/** map state to props */
const mapStateToProps = (
  state: Partial<Store>,
  ownProps: RouteComponentProps<RouteParams>
): DispatchedStateProps => {
  const searchedTitle = getQueryParams(ownProps.location)[QUERY_PARAM_TITLE] as string;
  const SMCPlansArray = makeSMCPlansArraySelector()(state, {
    plan_title: searchedTitle,
    statusList: allowedPlanStatusList,
  });

  return {
    ...ownProps,
    plans: SMCPlansArray,
  };
};

/** map dispatch to props */
const mapDispatchToProps = { fetchPlans: fetchSMCPlans };

/** Connected ActiveFI component */
const ConnectedSMCPlansList = connect(mapStateToProps, mapDispatchToProps)(SMCPlansList);

export default ConnectedSMCPlansList;
