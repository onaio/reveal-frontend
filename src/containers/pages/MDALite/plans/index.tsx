import reducerRegistry from '@onaio/redux-reducer-registry';
import React from 'react';
import { connect } from 'react-redux';
import { RouteComponentProps } from 'react-router';
import { Store } from 'redux';
import {
  HIDDEN_PLAN_STATUSES,
  SUPERSET_MDA_LITE_REPORTING_PLANS_SLICE,
} from '../../../../configs/env';
import { MDA_LITE_PLANS } from '../../../../configs/lang';
import { QUERY_PARAM_TITLE, REPORT_MDA_LITE_PLAN_URL } from '../../../../constants';
import { getPlanStatusToDisplay, getQueryParams, RouteParams } from '../../../../helpers/utils';
import supersetFetch from '../../../../services/superset';
import GenericPlansReducer, {
  genericFetchPlans,
  GenericPlan,
  makeGenericPlansArraySelector,
  reducerName as genericReducerName,
} from '../../../../store/ducks/generic/plans';
import { InterventionType } from '../../../../store/ducks/plans';
import { GenericPlanListProps, GenericPlansList } from '../../GenericPlansList';

/** register the MDA plan definitions reducer */
reducerRegistry.register(genericReducerName, GenericPlansReducer);

/** a list of plan statuses to be displayed */
const allowedPlanStatusList = getPlanStatusToDisplay(HIDDEN_PLAN_STATUSES);

/** selector for MDA lite plans */
const makeMDALitePlansArraySelector = makeGenericPlansArraySelector();

/** Simple component that loads a preview list of MDA plans */
const MDALitePlansList = (props: GenericPlanListProps & RouteComponentProps) => {
  return (
    <div>
      <GenericPlansList {...props} />
    </div>
  );
};

/** Declare default props for MDALitePlansList */
const defaultProps: GenericPlanListProps = {
  fetchPlans: genericFetchPlans,
  pageTitle: MDA_LITE_PLANS,
  pageUrl: REPORT_MDA_LITE_PLAN_URL,
  plans: [],
  service: supersetFetch,
  supersetReportingSlice: SUPERSET_MDA_LITE_REPORTING_PLANS_SLICE,
};

MDALitePlansList.defaultProps = defaultProps;

export { MDALitePlansList };

/** Connect the component to the store */

/** interface to describe props from mapStateToProps */
interface DispatchedStateProps extends RouteComponentProps<RouteParams> {
  plans: GenericPlan[];
}

/** map state to props */
const mapStateToProps = (
  state: Partial<Store>,
  ownProps: RouteComponentProps<RouteParams>
): DispatchedStateProps => {
  const searchedTitle = getQueryParams(ownProps.location)[QUERY_PARAM_TITLE] as string;
  const MDALitePlansArray = makeMDALitePlansArraySelector(state, {
    interventionTypes: [InterventionType.MDALite],
    plan_title: searchedTitle,
    statusList: allowedPlanStatusList,
  });

  return {
    ...ownProps,
    plans: MDALitePlansArray,
  };
};

/** map dispatch to props */
const mapDispatchToProps = { fetchPlans: genericFetchPlans };

/** Connected ActiveFI component */
const ConnectedMDALitePlansList = connect(mapStateToProps, mapDispatchToProps)(MDALitePlansList);

export default ConnectedMDALitePlansList;
