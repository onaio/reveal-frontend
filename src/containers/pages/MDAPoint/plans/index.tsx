import reducerRegistry from '@onaio/redux-reducer-registry';
import React from 'react';
import { connect } from 'react-redux';
import { RouteComponentProps } from 'react-router';
import { Store } from 'redux';
import {
  HIDDEN_PLAN_STATUSES,
  SUPERSET_MDA_POINT_REPORTING_PLANS_SLICE,
} from '../../../../configs/env';
import { MDA_POINT_PLANS } from '../../../../configs/lang';
import { QUERY_PARAM_TITLE, REPORT_MDA_POINT_PLAN_URL } from '../../../../constants';
import { getPlanStatusToDisplay, getQueryParams } from '../../../../helpers/utils';
import { RouteParams } from '../../../../helpers/utils';
import supersetFetch from '../../../../services/superset';
import GenericPlansReducer, {
  genericFetchPlans,
  GenericPlan,
  makeGenericPlansArraySelector,
  reducerName as genericReducerName,
} from '../../../../store/ducks/generic/plans';
import { InterventionType } from '../../../../store/ducks/plans';
import { GenericPlanListProps, GenericPlansList } from '../../GenericPlansList';

/** register the MDA plan definitions reducer */
reducerRegistry.register(genericReducerName, GenericPlansReducer);

/** a list of plan statuses to be displayed */
const allowedPlanStatusList = getPlanStatusToDisplay(HIDDEN_PLAN_STATUSES);

/** selector for MDA point plans */
const makeMDAPointPlansArraySelector = makeGenericPlansArraySelector();

/** Simple component that loads a preview list of MDA plans */
const MDAPointPlansList = (props: GenericPlanListProps & RouteComponentProps) => {
  return (
    <div>
      <GenericPlansList {...props} />
    </div>
  );
};

/** Declare default props for MDAPointPlansList */
const defaultProps: GenericPlanListProps = {
  fetchPlans: genericFetchPlans,
  pageTitle: MDA_POINT_PLANS,
  pageUrl: REPORT_MDA_POINT_PLAN_URL,
  plans: [],
  service: supersetFetch,
  supersetReportingSlice: SUPERSET_MDA_POINT_REPORTING_PLANS_SLICE,
};

MDAPointPlansList.defaultProps = defaultProps;

export { MDAPointPlansList };

/** Connect the component to the store */

/** interface to describe props from mapStateToProps */
interface DispatchedStateProps extends RouteComponentProps<RouteParams> {
  plans: GenericPlan[];
}

/** map state to props */
const mapStateToProps = (
  state: Partial<Store>,
  ownProps: RouteComponentProps<RouteParams>
): DispatchedStateProps => {
  const searchedTitle = getQueryParams(ownProps.location)[QUERY_PARAM_TITLE] as string;
  const MDAPointPlansArray = makeMDAPointPlansArraySelector(state, {
    interventionTypes: [InterventionType.MDAPoint],
    plan_title: searchedTitle,
    statusList: allowedPlanStatusList,
  });

  return {
    ...ownProps,
    plans: MDAPointPlansArray,
  };
};

/** map dispatch to props */
const mapDispatchToProps = { fetchPlans: genericFetchPlans };

/** Connected ActiveFI component */
const ConnectedMDAPointPlansList = connect(mapStateToProps, mapDispatchToProps)(MDAPointPlansList);

export default ConnectedMDAPointPlansList;
