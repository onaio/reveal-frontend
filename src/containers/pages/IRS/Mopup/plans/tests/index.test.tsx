import { mount, shallow } from 'enzyme';
import { createBrowserHistory } from 'history';
import React from 'react';
import { act } from 'react-dom/test-utils';
import { Provider } from 'react-redux';
import { Router } from 'react-router';
import { IRSMopUpReporting } from '..';
import { MOP_UP_REPORTING_TITLE } from '../../../../../../configs/lang';
import { IRS_MOP_UP_REPORT_URL } from '../../../../../../constants';
import store from '../../../../../../store';
import { GenericPlan } from '../../../../../../store/ducks/generic/plans';
import { mopupPlans } from '../../reports/tests/fixtures/fixtures';

jest.mock('../../../../../../configs/env');

const history = createBrowserHistory();

describe('IRS mop-up plans list', () => {
  beforeEach(() => {
    jest.resetAllMocks();
  });

  it('renders without crashing', () => {
    const props = {
      history,
      location: {
        hash: '',
        pathname: IRS_MOP_UP_REPORT_URL,
        search: '',
        state: undefined,
      },
      match: {
        isExact: true,
        params: {},
        path: `${IRS_MOP_UP_REPORT_URL}/`,
        url: `${IRS_MOP_UP_REPORT_URL}/`,
      },
      plans: mopupPlans as GenericPlan[],
    };
    shallow(
      <Router history={history}>
        <IRSMopUpReporting {...props} />
      </Router>
    );
  });

  it('renders plans correctly when connected to store', async () => {
    const service = jest
      .fn()
      .mockImplementationOnce(() => Promise.resolve(mopupPlans as GenericPlan[]));
    const props = {
      fetchPlans: jest.fn(),
      history,
      location: {
        hash: '',
        pathname: IRS_MOP_UP_REPORT_URL,
        search: '',
        state: undefined,
      },
      match: {
        isExact: true,
        params: {},
        path: `${IRS_MOP_UP_REPORT_URL}`,
        url: `${IRS_MOP_UP_REPORT_URL}`,
      },
      service,
    };
    const wrapper = mount(
      <Provider store={store}>
        <Router history={history}>
          <IRSMopUpReporting {...props} />
        </Router>
      </Provider>
    );
    await act(async () => {
      await new Promise(resolve => setImmediate(resolve));
      wrapper.update();
    });

    expect(wrapper.find('h3.page-title').text()).toEqual(MOP_UP_REPORTING_TITLE);

    expect(wrapper.find('DrillDownTable').text()).toMatchSnapshot();

    expect(wrapper.find('GenericPlansList').length).toBe(1);
  });
});
