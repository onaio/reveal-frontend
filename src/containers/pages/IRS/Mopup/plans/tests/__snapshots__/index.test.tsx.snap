// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IRS mop-up report renders plans correctly when connected to store 1`] = `"Row Height Row HeightdefaultshorttallCustomize ColumnsCustomize ColumnsSelect columns to be displayedTitleDate CreatedStart DateEnd DateStatusRows to display10203050Previouspage    of 4Next TitleDate CreatedStart DateEnd DateStatusVL Zambia 2020 Traning Plan2020-09-192020-09-192020-10-09activeDynamic-IRS 2020-09-192020-09-192020-09-192020-10-09activeixion - Dynamic-IRS 2020-09-192020-09-192020-09-192020-10-09activeVL Users Plan - Dynamic-IRS 2020-09-172020-09-172020-09-172020-10-07activeCR TEST IRS 2020-09-172020-09-172020-09-172020-10-07activeFO Nchelenge2020-09-162020-09-292020-10-03activeFO Chadiza2020-09-162020-09-162020-10-06activeDynamic-IRS 2020-09-15 Test RMK2020-09-152020-09-152020-10-05activeFO Training local - Dynamic-IRS 2020-09-152020-09-152020-09-152020-10-05activeFO Training - VL Hierarchy - Dynamic-IRS 2020-09-142020-09-142020-09-142020-10-04activeRows to display10203050Previouspage    of 4Next "`;
